### ISSUE TITLE
[bug] loss of inbound posts from Akkoma (and probably Pleroma)

### ISSUE THREAD

**Issue by mirabilos** [superseriousbusiness/gotosocial#2496]
### Describe the bug with a clear and concise description of what the bug is.

My SO is on an Akkoma instance, and I’ve put her account on notification, but still sometimes I cannot see posts she tells me she wrote, including Direct ones which `@` me.

Others then show up.

### What's your GoToSocial Version?

0.13.0-rc2 git-d0bb8f0

### GoToSocial Arch

amd64 binary

**Comment by tsmethurst** [comment]
Some logs would be really useful here, specifically, any logs that are shown on your instance and on hers when a message fails to be received. Without logs it's pretty hard to debug this.

**Comment by mirabilos** [comment]
tobi dixit:

>Without logs it's pretty hard to debug this.

I understand, but that’s hard to get:

>Some logs would be really useful here, specifically, any logs that are
>shown on your instance

We maybe can try to force this, sending test messages until I fail to
get one, but normally we only notice that by accident.

>and on hers when a message fails to be received.

She’s on an open instance, so unfortunately no logs from there.
(infosec.place, Akkoma 3.10.4-1-gebfb617b)

I thought to file this anyway, perhaps someone has an idea or something.

bye,
//mirabilos

**Comment by mirabilos** [comment]
I wonder if this is not related to #2460, although this case is a bit special in that I’m actively @-ed in the missing posts, whereas that is more about thread context.

Or, or and/or, maybe related to #2327?

Oh well, I’ll see once we have a release candidate…

**Comment by mirabilos** [comment]
Unfortunately, this is getting worse. I’m losing lots of incoming direct messages (which are in threads).

Regarding #2327, it also seems to be getting worse, I cannot resolve lots of polymaths threads’ context any more.

**Comment by tsmethurst** [comment]
Isn't polymath a gts instance? I don't think https://github.com/superseriousbusiness/gotosocial/issues/2327 is relevant there.

Anyways, we still can't do much to debug this without any logs or anything. Do you see anything logged in your GtS instance when you know a missing message has been sent? That might help us to find the problem.

Also, we reworked a lot of threading and thread deref logic on main, so whatever is causing the problem, it's quite possible it will be fixed by the changes we already made. If you don't want to run a snapshot from main, no problem, but it'll be a couple weeks til the first RC is out.

**Comment by mirabilos** [comment]
Unfortunately, I can currently only guess when posts are lost.

I might try running a snapshot if there will be an upgrade path.

**Comment by tsmethurst** [comment]
Is this still a problem or can I close the issue? I haven't seen anyone else mentioning it so just wanted to check.

**Comment by mirabilos** [comment]
tobi dixit:

>Is this still a problem or can I close the issue? I haven't seen anyone
>else mentioning it so just wanted to check.

We’ve not noticed it for a while, but it’s hard to notice.
I think closing it at this point is appropriate.

Thanks anyway!

**Comment by mirabilos** [comment]
We _might_ want to reopen this, but it’s a wider problem than just GtS, so I have no idea where to best report this.

Today, I managed to debug this a bit further.

The communication partners in the first part of this are:

- ME = me on GtS 0.16.0rc3
- SO = sender on Akkoma 3.13.2
- TP = third party on Mastodon 4.2.10

What happened was:

- SO sent a DM to ME and TP
- I got it, went to reply in Semaphore and saw only `@SO` in the textarea and added `@TP` manually
- TP asked what post I was referring to, as he did not receive the thread starter from SO

At this point I begun debugging. Looking into the GtS DB showed me that while there are two `<a class="u-url mention" …>` links in the `content` column, the `mentions` column has an array with just one ULID.

(At this point, SO tried to edit the post, adding *another* `@TP` mention, but while the textual change showed up, neither did it show up in the `mentions` array nor did TP become able to see it.)

I did a couple database queries more. I looked in `instances` which instances all used the exact same Akkoma version (6 in total), then did a query across `statuses` for multiple `<a class="u-url mention"` but lower array size of `mentions`, but unlisted or public posts (so I could actually look at the AP with a shell script I have that can do signed GET requests).

A bit of fiddling later (turns out Akkoma requires the actor to have an `inbox` and `outbox`, unlike Mastodon, GtS and Pleroma), and I was able to look at a public Note from Akkoma… which had:

```
[…]
  "tag": [
    {
      "href": "https://stereophonic.space/users/newt",
      "name": "@newt@stereophonic.space",
      "type": "Mention"
    },
    {
      "href": "https://suya.place/users/a1ba",
      "name": "@a1ba@suya.place",
      "type": "Mention"
    }
  ],
[…]
```

So, two mentions. However, in the GtS database, there was only one mention! (The thing I can obviously *not* look at is the POST to my inbox which probably populated the GtS DB.)

Unfortunately, the operator of the Akkoma instance I found the public post on is not knowledgeable enough to debug it, and I’m not sure if the admin of SO’s instance is (they run multiple instances of multiple different instance softwares, so they’re probably too busy) plus debugging something with private messages is iffy.

I’ve no idea if there’s something we can do on the GtS side to debug this. I’ll open up an issue on Akkoma pointing here, so they can look at this, hopefully.
